### PR TITLE
👷🏻‍♀️🛠 Xcode 9.3

### PR DESCRIPTION
**Pull request by phatblat**
Updated Travis to build with [`xcode9.3`](https://docs.travis-ci.com/user/reference/osx/#OS-X-Version).

Also updated test dependencies and project settings. 

- ⬆️ Nimble (7.0.3)
- ⬆️ Quick (1.2.0)
- ⬆️ ZipArchive (2.1.2)
- ⬆️ xcconfigs (0.11+ jspahrsummers/xcconfigs@bb795558a76e5daf3688500055bbcfe243bffa8d)

Lots of nullability warnings were breaking the build so I relaxed xcconfig's "treat warnings as errors". ~~I'd like to circle back and clean these up.~~ @tiennou and I cleaned up the nullability warnings. Note that these warnings were disabled in the test targets to keep the test code more lean.

### CODE REVIEW

File: ObjectiveGitTests/QuickSpec+GTFixtures.m [hunk 1]
```diff
@@ -6,10 +6,12 @@
 //  Copyright (c) 2013 GitHub, Inc. All rights reserved.
 //
 
-#import <ObjectiveGit/ObjectiveGit.h>
 #import "QuickSpec+GTFixtures.h"
-#import <objc/runtime.h>
-#import <ZipArchive/ZipArchive.h>
+//#import <objc/runtime.h>
```
Comment (tiennou) [inline ObjectiveGitTests/QuickSpec+GTFixtures.m]: Leftover ?

Comment (phatblat) [reply ObjectiveGitTests/QuickSpec+GTFixtures.m]: Yeah, I’ll clean that up

**Comment by phatblat** [comment]
Thanks for the nullable cleanup, @tiennou! I'll see if I can finish it off.

**Comment by phatblat** [comment]
Travis now has `xcode9.3`

**Pull request merged**